11 - Docker image and ROS2 installation in Ubuntu/GNU-Linux

mxochicale opened the pull request:
## Description
<!-- Replace {issue_number} with the issue that will be closed after merging this PR -->
Fixes #11 

## Type of change
Please delete options accordingly to the description.

<!-- Write an `x` in all the boxes that apply -->
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update


### Suggested Checklist
<!-- You do not need to complete all the items by the time you submit the pull request, but most likely the changes will only be merged if all the tasks are done. -->

<!-- Write an `x` in all the boxes that apply -->
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation 
- [x] My changes generate no new warnings 
- [x] I have commented my code, particularly in hard-to-understand areas 
- [x] I have read the [`CONTRIBUTING`](https://github.com/sense-base/base/blob/main/CONTRIBUTING.md) docs
- [x] My code passes, following the style guidelines `pre-commit run -a` 
- [x] My code is properly tested with `pytest -sv tests`
- [x] I have requested review to this PR.
- [x] I have addressed and marked as resolved all the review comments in my PR.
- [x] Make sure your branch is up-to-date with main branch. See step 4 in [`CONTRIBUTING`](https://github.com/sense-base/base/blob/main/CONTRIBUTING.md#pull-request-pr-and-merge-to-main-branch).
- [x] Finally, I have selected `squash and merge`

nikk-nikaznan commented on .devcontainer/README.md:
````diff
@@ -0,0 +1,52 @@
+### Build docker image
+* Building estimated time will depend on internet speed connetion
+```
+docker compose -f docker-compose.yml build # 1359.8s (13/13) FINISHED
````
```suggestion
docker compose -f docker-compose.yml build
```

mxochicale commented:
Thanks @nikk-nikaznan , let's catch up later after our meeting. Feel free to add logs for mac setup up and your chip in respective issue or PRs so we can track it. I am not sure if we expect mac users so that's a good question when we meet all the team :)

ahmed-alhindawi commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,112 @@
+    ros-${ROS_DISTRO}-diagnostic-updater \
+    ros-${ROS_DISTRO}-robot-localization \
+    ros-${ROS_DISTRO}-common-interfaces && \
+    rm -rf /var/lib/apt/lists/*
```
To tackle #4 , we can also install `ros-${ROS_DISTRO}-rmw-cyclonedds-cpp` as part of the ROS installation

mxochicale commented:
Thanks @ahmed-alhindawi, just added your suggestion and other minor changes to documentation. Let us know if you have other suggestions.

We removed the commented lines that we will incorporate future PRs when we need them :) 
```
# COPY *neural_depth_2.0_2080.model /usr/local/zed/resources/neural_depth_2.0.model
# COPY *.8600_timing_cache /usr/local/zed/resources/.8600_timing_cache
# COPY *.neural_depth_2.0.model_optimized-fbcbl-1-75-12020-8801-8600-25-384-6144-48-64-512-72-0-0e53-512 /usr/local/zed/resources/.neural_depth_2.0.model_optimized-fbcbl-1-75-12020-8801-8600-25-384-6144-48-64-512-72-0-0e53-512
# COPY *SN21837864.conf /usr/local/zed/settings/


# Install gazebo
# RUN wget https://packages.osrfoundation.org/gazebo.gpg -O /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg \
#   && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/pkgs-osrf-archive-keyring.gpg] http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/gazebo-stable.list > /dev/null \
#   && apt-get update && apt-get install -q -y --no-install-recommends \
#     ros-iron-gazebo* \
#   && rm -rf /var/lib/apt/lists/*
```

nikk-nikaznan commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,101 @@
+    ros-${ROS_DISTRO}-robot-localization \
+    ros-${ROS_DISTRO}-common-interfaces \
+    ros-${ROS_DISTRO}-rmw-cyclonedds-cpp && \
+    # ros-${ROS_DISTRO}-common-interfaces && \
```
Should we remove this for now?

nikk-nikaznan commented on docs/docker/launch_image.bash:
```diff
@@ -0,0 +1,14 @@
+XSOCK=/tmp/.X11-unix
+
+docker run -it \
+ --gpus all \
```
```suggestion
  --runtime=nvidia \
```

nikk-nikaznan reviewed (approved):
I can build the container both ways now `devcontainer` and `terminal`

nikk-nikaznan commented on docs/docker/README.md:
````diff
@@ -0,0 +1,60 @@
+
+### Launch and test docker image
+```
+bash launch_image.bash
````
```suggestion
bash docs/docker/launch_image.bash
```

nikk-nikaznan commented on docs/docker/README.md:
````diff
@@ -0,0 +1,60 @@
+
+### Stop container and remove it
+```
+bash stop_container_and_removeit.bash
````
```suggestion
bash docs/docker/stop_container_and_removeit.bash
```
